week-5/endpoints.ts

```diff
@@ -0,0 +1,106 @@
+import * as url from 'url';
+import * as http from 'http';
+import UserModel from './UserModel';
+import type { HobbiesResponse, PostUser, UserResponse, SuccessfulResponse } from './types';
+
+const domain = "http://localhost:8000";
+
+export const processGetRequest = (userModel: UserModel, req: http.IncomingMessage): HobbiesResponse | undefined | UserResponse[] => {
+  const url = req.url;
+  if(url === undefined ) {
+    return undefined;
+  }
+  const regexp = /^\/api\/users\/(?<userid>[a-zA-Z0-9\-]+)\/hobbies$/ig;
+  if(url === '/api/users'){
+    return userModel.getUsers();
+  } else {
+    for(const match of url.matchAll(regexp)) {
+      if(match.groups !== undefined) {
+        return userModel.getHobbiesResponse(match.groups.userid);
+      } else {
+        return undefined;
+      }
+    }
+  }
+  return undefined;
+};
+
+export const processPostRequest = (userModel: UserModel, req: http.IncomingMessage) => new Promise((resolve, reject) => {
+  const url = req.url;
+  if (url === undefined) {
+    resolve(undefined);
+  }
+  if(url === '/api/users') {
+    parseRequestBody(req).then((parsedBody) => {
+      const newUser = userModel.createUser(parsedBody as PostUser);
+      resolve(newUser);
+    })
+    .catch((error: Error) => {
+      reject(error);
+    });
+  } else {
+    resolve(undefined);
+  }
+});
```

Comment:
I am curious why you used an explicit Promise instead of async/await. By using async/await the flow of the function can be simplified
```suggestion
export const processPostRequest = async (
  userModel: UserModel,
  req: http.IncomingMessage
) => {
  const url = req.url;

  if (url === undefined) {
    return;
  }

  try {
    if (url === "/api/users") {
      const parsedBody = await parseRequestBody(req);

      return userModel.createUser(parsedBody as PostUser);
    }
  } catch (error) {
    throw error;
  }

  return;
};
```

Reply:
The reason was, I tried but for some reason didn't work well that time, so I decided to write a plain promise instead.
Thanks for the suggestion, I applied it to two functions I had.